config/routes.rb

```diff
@@ -1,4 +1,35 @@
 Rails.application.routes.draw do
+
+  root "welcome#index"
+
+  get 'welcome/index'
+
+  resources :products, :only => [:index, :show] do
+    resources :order_items, :only => [:create, :update, :destroy]
+    resources :reviews
+    post 'add_to_cart', on: :member
+  end
+  get 'search' => 'products#search', as: 'search_product'
+
+  resources :users, :only => [:new, :create, :show, :edit, :update] do
+    resources :reviews
+    scope module: :users do
+      resources :products
+      resources :fulfillment, :only => [:index, :show]
+    end
+  end
+
+  resources :orders, :only => [:show, :edit, :update]
+
+  resources :order_items, :only => [:show]
+
+  resources :sessions, :only => [:create]
+  delete "/logout" => "sessions#destroy"
+  get    "/login"  => "sessions#new"
+
+  resources :categories, :only => [:new, :create]
+
+
```

Comment:
Totally get it. Ada is a whirlwind. With code reviews I try to do a scattershot comment method so that folks can just let go the things they would have done if they had time and focus on seeing the things they didn't think of. You will always run out of time, it is the way that it goes. When you are employed you will be surprised how the pace slows down. The requirements for high fidelity rise, which is part of it, but it is part of why Ada students seem so amazing to folks on the outside. Because the pace of coding is actually super fast.